src/uwtools/api/config.py

```diff
@@ -94,6 +94,7 @@ def get_yaml_config(config: Union[dict, Optional[Path]] = None) -> _YAMLConfig:
 def realize(
     input_config: Union[dict, _Config, Optional[Path]] = None,
     input_format: Optional[str] = None,
+    output_block: Optional[List[Union[str, int]]] = None,
```

Comment:
Per the YAML spec (https://yaml.org/spec/1.2.2/#mapping, section 10.1.1.1)
> YAML places no restrictions on the type of [keys](https://yaml.org/spec/1.2.2/#nodes); in particular, they are not restricted to being [scalars](https://yaml.org/spec/1.2.2/#scalars).

So we might consider expanding the type of this argument. The `uwtools.config.jinja2` module has a `_ConfigVal` custom type that could maybe be copied and pared down (removing the `UWYAMLConvert` and `UWYAMLRemove` types) for this purpose.

Reply:
I'm not even entirely sure I still support my initial call to include `int` in the options here. Despite the wide open options that YAML has, I'm not entirely sure UW will want to support all of those. We found that allowing `int` keys leads to problems when we try to expand the dict's `a` and `b` with this sort of syntax:

```
{**a, **b}
```

Namespace becomes a real issue with boolean values. We won't be able to (I don't think) navigate to a list/dict entry in the YAML from the command line, and I'm not entirely sure what that even means to have those objects as keys. 

Perhaps we do want to limit UW YAML to scalar strings?

Reply:
All good points. I think that string-only keys would be a good convention to try for. If something comes up that makes that infeasible, I'm sure we can figure something out, but it's a sane starting position.